application/src/main/java/org/openhds/mobile/activity/NavigateActivity.java

```diff
@@ -322,12 +340,48 @@ private void hierarchySetup() {
             showDetailFragment();
             detailToggleFragment.setButtonHighlighted(true);
         }
+        //populate recent forms
+        populateFormsForPath();
 
         if(null != getCurrentVisit()){
             visitFragment.setButtonEnabled(true);
         }
     }
 
+    private void populateFormsForPath() {
+
+        List<FormInstance> recforms;
+        ContentResolver reslovr = this.getContentResolver();
+        DatabaseAdapter dbAdpt = new DatabaseAdapter(this);
+
+        //get all the existing recent form paths
+        Collection<String> formsdel=  dbAdpt.findAllFormPaths();
+
+        //find if any of the form is submitted
+        List <String> submitForms  = OdkCollectHelper.getSentFormPaths(getContentResolver(), formsdel);
+
```

Comment:
The formatting in the code base isn't consistent, but we should try not to make it worse. These lines have odd whitespace. You can select the blocks you add and run the code formatter on them.